beacon-chain/db/filesystem/pruner.go

```diff
@@ -93,13 +93,15 @@ func windowMin(latest, offset primitives.Slot) primitives.Slot {
 func (p *blobPruner) warmCache() error {
 	p.Lock()
 	defer p.Unlock()
+	defer func() {
```

Comment:
does the order of the defer above for unlock matter? I suppose if it unlocks here this other defer is updating p.warmed without the lock?

Reply:
defers are always LIFO, so this code is correct (unlock happens after the channel is closed). But, I did anticipate this might confuse someone and thought about putting the unlock in the other deferred function, so I think I'll go ahead and do that.